examples/others/tensorize_vllm_model.py

```
@@ -298,30 +307,36 @@ def deserialize():
         else:
             model_path = f"{base_path}/model.tensors"
 
-        tensorizer_config = TensorizerConfig(
-            tensorizer_uri=model_path,
-            encryption_keyfile=keyfile,
-            **credentials)
-
-        if args.serialization_kwargs:
-            serialization_kwargs = json.loads(args.serialization_kwargs)
-            tensorizer_config.serialization_kwargs = serialization_kwargs
-            print("Found serialization kwargs: ", serialization_kwargs)
 
+        if not tensorizer_args:
+            tensorizer_config = TensorizerConfig(
+                tensorizer_uri=model_path,
+                encryption_keyfile=keyfile,
+                **credentials
+            )
+            if args.serialization_kwargs:
+                serialization_kwargs = json.loads(args.serialization_kwargs)
+                tensorizer_config.serialization_kwargs = serialization_kwargs
 
-        if args.lora_path:
-            tensorizer_config.lora_dir = tensorizer_config.tensorizer_dir
-            tensorize_lora_adapter(args.lora_path, tensorizer_config)
+            if args.lora_path:
+                tensorizer_config.lora_dir = tensorizer_config.tensorizer_dir
+                tensorize_lora_adapter(args.lora_path, tensorizer_config)
```

It doesn't seem appropriate to ignore all of these if both they and `tensorizer_args` exist. If it is specified both ways, it should either simply raise an error telling you to pick one way, or merge the settings.

---

Good call. I'll have the `--model-loader-extra-config` JSON take precedence in updating values.